lib/logger.ts

```diff
@@ -0,0 +1,43 @@
+import winston = require('winston');
+import DailyRotateFile = require('winston-daily-rotate-file');
+
+const logLevel = process.env.LOG_LEVEL || 'info';
+const fileName = process.env.LOG_FILE_NAME || './logs/node-sdk-core.log';
+let loggerDisabled = true;
+let fileTransport;
+
+// enable logger if any transport is set
+if (process.env.ENABLE_ROTATING_FILE === 'true' || process.env.LOG_TO_CONSOLE === 'true' ||
+process.env.LOG_FILE_NAME) {
+  loggerDisabled = false;
+}
+
+const logger = winston.createLogger({
+  level: logLevel,
+  format: winston.format.json(),
+  silent: loggerDisabled
+});
+
+// If flag is enabled then log to the `console`
+if (process.env.LOG_TO_CONSOLE === 'true') {
+  logger.add(new winston.transports.Console());
+}
+// If LOG_FILE_NAME is set, then log to specified file
+if (process.env.LOG_FILE_NAME && !process.env.ENABLE_ROTATING_FILE) {
+  fileTransport = new winston.transports.File({ filename: fileName });
+  logger.add(fileTransport);
+}
+// If flag is enabled then remove existing file transport & add rotating log file transport
+else if (process.env.ENABLE_ROTATING_FILE === 'true') {
+  if (fileTransport) {
+    logger.remove(fileTransport);
+  }
```

Comment:
Correct me if I'm wrong, but this block seems like it will never execute thanks to the if/else that this is wrapped in so it probably isn't necessary

Reply:
Nope, you are right. Good catch, I originally had it set to just two if statements